kolibri/plugins/management/assets/src/state/actions.js

```diff
@@ -373,77 +375,103 @@ function createUser(store, stateUserData) {
 
 /**
  * Do a PATCH to update existing user
- * @param {object} stateUser
- * Needed: id
- * Optional Changes: full_name, username, password, facility, kind(role)
+ * @param {object} store
+ * @param {string} userId
+ * @param {object} userUpdates Optional Changes: full_name, username, password, and kind(role)
  */
-function updateUser(store, stateUser) {
-  // payload needs username, fullname, and facility
-  const savedUserModel = FacilityUserResource.getModel(stateUser.id);
+function updateUser(store, userId, userUpdates) {
+  store.dispatch('SET_ERROR', '');
+  store.dispatch('SET_BUSY', true);
+  const savedUserModel = FacilityUserResource.getModel(userId);
   const savedUser = savedUserModel.attributes;
-  const changedValues = {};
-  let roleAssigned = Promise.resolve(savedUser);
 
   // explicit checks for the only values that can be changed
-  if (stateUser.full_name && stateUser.full_name !== savedUser.full_name) {
-    changedValues.full_name = stateUser.full_name;
+  const origUserState = store.state.pageState.facilityUsers.find(user => user.id === userId);
+  const changedValues = {};
+  if (userUpdates.full_name && userUpdates.full_name !== origUserState.full_name) {
+    changedValues.full_name = userUpdates.full_name;
   }
-  if (stateUser.username && stateUser.username !== savedUser.username) {
-    changedValues.username = stateUser.username;
+  if (userUpdates.username && userUpdates.username !== origUserState.username) {
+    changedValues.username = userUpdates.username;
   }
-  if (stateUser.password && stateUser.password !== savedUser.password) {
-    changedValues.password = stateUser.password;
+  if (userUpdates.password && userUpdates.password !== origUserState.password) {
+    changedValues.password = userUpdates.password;
+  }
+  if (userUpdates.kind && userUpdates.kind !== origUserState.kind) {
+    changedValues.kind = userUpdates.kind;
   }
 
-  if (stateUser.kind && stateUser.kind !== _userState(savedUser).kind) {
-    // assumes there's no previous roles to delete at first
-    let handlePreviousRoles = Promise.resolve();
+  if (Object.getOwnPropertyNames(changedValues).length === 0) {
+    displayModal(store, false);
+    store.dispatch('SET_ERROR', '');
+    store.dispatch('SET_BUSY', false);
+  } else {
+    let roleAssigned = Promise.resolve(savedUser);
+
+    if (changedValues.kind) {
+      // assumes there's no previous roles to delete at first
+      let handlePreviousRoles = Promise.resolve();
+
+      if (savedUser.roles.length) {
+        const roleDeletes = [];
+        savedUser.roles.forEach(role => {
+          roleDeletes.push(RoleResource.getModel(role.id).delete());
+        });
```

Comment:
You could `map` over `roles`

```javascript
const roleDeletes = savedUser.roles.map((role) => RoleResource.getModel(role.id).delete())
```

Not sure, but might also be possible to just eliminate the `if` block entirely, since `Promise.all([])` sort of works like `Promise.resolve()`